AGENTS MD: Add suggestion on how to incorporate tests (#47784)
vllm-project/vllm: AGENTS.md

uv pip install -e . --torch-backend=auto
```

### Tests

> Requires [Environment setup](#environment-setup) and [Installing dependencies](#installing-dependencies).

```bash
# Install test dependencies (use cuda.in on non-x86_64):
uv pip install -r requirements/test/cuda.in

# Run a specific test file:
.venv/bin/python -m pytest tests/path/to/test_file.py -v
```

When adding tests:

> 

Reviewer: ```suggestion

- **Design before you write.** Answer four questions first: what is the module for, what is its I/O contract, what failure am I guarding against, and what is the cheapest level that catches it (unit over integration over e2e)?
```

- **Reuse before create.** Extend existing test files, `conftest.py` fixtures, and
  helpers; add a new file only when no nearby suite fits.
- **Test behavior, not structure.** Assert observable outcomes through public
  APIs; skip trivial wiring and implementation-locked checks.
- **Every test needs a reason.** Prefer regression tests and paths that break

>   easily; flaky tests are worse than no tests.

Reviewer: ```suggestion 
- **Every test needs a reason.** Prefer regression tests and paths that break easily, and state the reason in the test name or docstring; flaky tests are worse than no tests.
```

- **No one-off kernel benchmarks in `tests/`.** Put kernel perf work in
  `benchmarks/kernels/`; prove correctness in existing pytest suites.
- **Run model evals for model-affecting changes.** Search `tests/evals/` or use
  `vllm bench` and include results in the PR — do not wait for reviewers to ask.

> 

Reviewer: ```suggestion
- **Keep it minimal.** One behavior per test and the smallest setup that triggers it; if the test diff dwarfs the code change, cut scope.
- **Stay realistic.** Pull shapes, dtypes, and configs from shipped models (a Qwen head size, a real block size); parametrize a representative set, not the full cross-product.
- **Make it deterministic.** Seed everything, use greedy decoding for output checks, and compare floats with dtype-appropriate tolerances, never exact match.
- **Respect the CI budget.** Use the smallest model that exercises the path, keep runtime in seconds, and gate anything heavier behind nightly or opt-in markers.
- **Do not land permanent debt.** No golden outputs that rot with model updates, no hardcoded environment assumptions, no skip/xfail without a linked issue.
- 

```

Author: thanks. we also need to keep AGENTS.md short so can we pick only one or two edits?

Reviewer: I would say that these two are the most important ones then:
```
- **Design before you write.** Answer four questions first: what is the module for, what is its I/O contract, what failure am I guarding against, and what is the cheapest level that catches it (unit over integration over e2e)?
- **Keep it minimal.** One behavior per test and the smallest setup that triggers it; if the test diff dwarfs the code change, cut scope.
```

### Coding style guidelines

> - Match existing code style; keep comments and docstrings minimal.

Reviewer 2: ```suggestion
- Match existing code style
- Minimize use of comments. Eliminate comments which are redundant, preferring legible and self-documenting code. When used, keep docstrings and comments brief and direct.
```
is this too much?
